src/db/oracledb/jobs-dao.js

```diff
@@ -37,23 +120,316 @@ const getJobs = async (internalId, query) => {
+ */
+const graduateCreateJob = async (connection, osuId, body) => {
+  const binds = standardBinds(osuId, body, graduateBinds);
+
+  const { outBinds: { result } } = await connection.execute(
+    contrib.graduateCreateJob(binds),
+    binds,
+  );
+  return result;
+};
+
+/**
+ * Updates graduate job records
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @returns {string} Query result, null if success
+ */
+const graduateUpdateJob = async (connection, osuId, body) => {
+  const binds = standardBinds(osuId, body, graduateBinds);
+
+  const { outBinds: { result } } = await connection.execute(
+    contrib.graduateUpdateJob(binds),
+    binds,
+  );
+  return result;
+};
+
+/**
+ * Determines if changeReasonCode is valid
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} changeReasonCode Change reason code from request body
+ * @returns {boolean} True if changeReasonCode is valid
+ */
+const isValidChangeReasonCode = async (connection, changeReasonCode) => {
+  const { rows } = await connection.execute(
+    contrib.validateChangeReasonCode(),
+    { changeReasonCode },
+  );
+  return rows[0].count > 0;
+};
+
+/**
+ * Determines which Epaf to execute based on fields in body
+ *
+ * @param {boolean} update True if updating record. False if creating record
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @param {string} internalId Internal ID of a person
+ * @returns {object} raw job record
+ */
+const createOrUpdateJob = async (update, osuId, body) => {
+  const connection = await getConnection('banner');
+  try {
+    let result;
+    const { studentEmployeeInd, changeReason: { code: changeReasonCode } } = body;
+    const employmentType = studentEmployeeInd ? 'student' : 'graduate';
 
-    if (rows.length > 1) {
-      throw new Error(`Multiple job records found for job ID ${jobId}`);
+    if (employmentType === 'student' && !/^C5[0-2]/.test(body.positionNumber)) {
+      return new Error('Position number for students must start with C50, C51, or C52');
     }
 
-    await getLaborDistributions(connection, internalId, rows);
+    if (_.includes(['TERME', 'TERMJ'], changeReasonCode)) {
+      result = await terminateJob(connection, osuId, body);
+    } else {
+      if (!await isValidChangeReasonCode(connection, changeReasonCode)) {
+        return new Error(`Invalid change reason code ${changeReasonCode}`);
+      }
 
-    return rows[0];
+      if (update) {
+        if (changeReasonCode === 'NONE') {
+          result = await updateLaborChangeJob(connection, osuId, body);
+        } else if (changeReasonCode === 'BREAP') {
+          if (employmentType === 'student') {
+            result = await studentUpdateJob(connection, osuId, body);
+          } else if (employmentType === 'graduate') {
+            result = await graduateUpdateJob(connection, osuId, body);
```

Comment:
`studentUpdateJob` and `graduateUpdateJob` are almost the same besides the contrib they call right? We can probably combine them into one `updateJob` and pass the `employmentType` as one argument to determine which contrib should be used. Same thing for `createJob`.